quickwit/quickwit-common/src/rendezvous_hasher.rs

```diff
@@ -21,20 +21,20 @@ use std::cmp::Reverse;
 use std::collections::hash_map::DefaultHasher;
 use std::hash::{Hash, Hasher};
 
-/// Computes the affinity of a node with a given `split_id`.
+/// Computes the affinity of a node with a given `key`.
 ///
 /// We rely on rendez-vous hashing here.
-fn node_affinity<T: Hash>(node: T, split_id: &str) -> u64 {
+fn node_affinity<T: Hash, U: Hash>(node: T, key: &U) -> u64 {
     let mut state = DefaultHasher::new();
-    split_id.hash(&mut state);
+    key.hash(&mut state);
     node.hash(&mut state);
     state.finish()
 }
 
 /// Sorts the list of node base on rendez-vous-hashing.
 /// Nodes are ordered by decreasing order of computed `hash_key`
-pub(crate) fn sort_by_rendez_vous_hash<T: Hash>(nodes: &mut [T], split_id: &str) {
-    nodes.sort_by_cached_key(|node| Reverse(node_affinity(node, split_id)));
+pub fn sort_by_rendez_vous_hash<T: Hash, U: Hash>(nodes: &mut [T], key: U) {
+    nodes.sort_by_cached_key(|node| Reverse(node_affinity(node, &key)));
```

Comment:
I know this is not part of your code, but do we actually care about the Reverse part here?
Do we use the node_affinity function somewhere else? 
```suggestion
    nodes.sort_by_cached_key(|node| node_affinity(node, &key));
```

Reply:
`node_affiinity` is used only here. And actually, I don't know why we put `Reverse` :/

Reply:
Mmmm, it seems that `node_affinity` is a bit misleading.
I would rename `node_affinity` by `rendez_vous_hash` and remove the `Reverse`.

Reply:
`Reverse` for highest affinity first:

```rust
for job in jobs {
    sort_by_rendez_vous_hash(&mut nodes, job.split_id());
    // choose one of the the first two nodes based on least loaded
    let chosen_node_index: usize = if nodes.len() >= 2 {
        usize::from(nodes[0].load > nodes[1].load)
    } else {
        0
    };
```

Reply:
I'm smoking t_t

Reply:
Also I'm wondering why we did not choose to node weights to deduce a score from hash + weigth.

Reply:
> did not choose to use node weights to deduce a score from hash + weigth

I am not entirely sure what this would mean.

Reply:
I think `node_affinity` is good. what makes it misleading?